x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -81,10 +82,36 @@ public class ReservedRolesStore implements BiConsumer<Set<String>, ActionListene
         null,
         new String[] { "*" },
         MetadataUtils.DEFAULT_RESERVED_METADATA,
-        Collections.emptyMap()
+        Collections.emptyMap(),
+        TcpTransport.isUntrustedRemoteClusterEnabled()
+            ? new RoleDescriptor.RemoteIndicesPrivileges[] {
+                new RoleDescriptor.RemoteIndicesPrivileges(
+                    RoleDescriptor.IndicesPrivileges.builder().indices("*").privileges("all").allowRestrictedIndices(false).build(),
+                    "*"
+                ),
+                new RoleDescriptor.RemoteIndicesPrivileges(
+                    RoleDescriptor.IndicesPrivileges.builder()
+                        .indices("*")
+                        .privileges("monitor", "read", "view_index_metadata", "read_cross_cluster")
+                        .allowRestrictedIndices(true)
+                        .build(),
+                    "*"
+                ) }
```

Comment:
We can discuss whether `superuser` should only have `read` and `read_cross_cluster` privileges. They are the only supported ones by the new RCS. But the remote indices privileges itself does not prevent the usage of any index privileges, e.g. it is totally ok to define a custom role that has `all` privileges for any remote indices. 

So my view is that for the privileges itself, it do not need to care whether certain privilege is supported by new RCS, which is a downstream implementation detail. For the superuser role, it is much easier to say that the role has exactly the same indices privileges on any clusters and this suits what a `superuser` would imply - a builtin powerful role that is identical everywhere. 

Bottom line is that we don't have any security concern since the ultimate permissions are subject to the cluster API key.

Reply:
I think mirroring the index privileges (i.e., including `all`) is clearer, and as you point out there are no security concerns here per se.

One thing this raises for me is whether we should (in the future) consider downgrading received privileges on the fulfilling cluster (e.g., `all` maps to `read`, `read_cross_cluster`) to more explicitly enforce the expectation that remote users only ever have read-access. Admins might get lazy and use a remote access API key that has `all`. 

This _should_ be unnecessary and may be overly complex, but it also seems like a good idea to explicitly enforce security assumptions. Certainly not for this PR, and not for now, just jotting this down as a thought.

Reply:
> I think mirroring the index privileges (i.e., including all) is clearer, and as you point out there are no security concerns here per se.

👍 

> whether we should (in the future) consider downgrading received privileges on the fulfilling cluster (e.g., all maps to read, read_cross_cluster) to more explicitly enforce the expectation that remote users only ever have read-access. Admins might get lazy and use a remote access API key that has all.

I don't think we should do that. It feels a bit overly smart. The remote access port only allows limited actions. So the privileges themselves are not useful without any actual actions. It is also hard to be complete because privileges can be assigned with actions,e.g. `all` is also `"indices:*", "internal:transport/proxy/indices:*"` and `indices:*` can also be replaced by `indices:admin*` and `indices:data*` and many other variations.